Sources/SwiftFusion/Inference/NewJacobianFactor.swift

```diff
@@ -66,11 +66,25 @@ public struct NewJacobianFactor<
   }
 
   public func errorVector_linearComponent(_ x: Variables) -> ErrorVector {
-    return ErrorVector(jacobian.lazy.map { $0.dot(x) })
+    var r = ErrorVector.zero
```

Comment:
This one surprises me, because in theory it writes on `r` twice. When forced to write something out the ugly way or with a raw loop, or especially with code that looks like it shold be inefficient, it's a good idea to put in a comment that explains both what it does, and why you wrote it that way so someone doesn't come along later and say “oh, that's a one-liner” and rewrite it.
```suggestion
    // ErrorVector(jacobian.lazy.map { $0.dot(x) }), optimized
    var r = ErrorVector.zero
```

Reply:
I figured out a way to get the oneliner with `map` to optimize well. I'll upload the commit soon.